packages/core/src/lib/history.unit.test.ts

```diff
@@ -0,0 +1,137 @@
+import { describe, expect, vi } from 'vitest';
+import { MINIMAL_CONFIG_MOCK } from '@code-pushup/test-utils';
+import { getCurrentBranchOrTag, safeCheckout } from '@code-pushup/utils';
+import { collectAndPersistReports } from './collect-and-persist';
+import { HistoryOptions, history, prepareHashes } from './history';
+import { upload } from './upload';
+
+vi.mock('@code-pushup/utils', async () => {
+  const utils: object = await vi.importActual('@code-pushup/utils');
+  let currentBranchOrTag = 'main';
+  return {
+    ...utils,
+    safeCheckout: vi.fn().mockImplementation((branch: string) => {
+      currentBranchOrTag = branch;
+    }),
+    getCurrentBranchOrTag: vi.fn().mockImplementation(() => currentBranchOrTag),
+  };
+});
+
+vi.mock('./collect-and-persist', () => ({
+  collectAndPersistReports: vi.fn(),
+}));
+
+vi.mock('./upload', () => ({
+  upload: vi.fn(),
+}));
+
+describe('history', () => {
+  it('should check out all passed commits and reset to initial branch or tag', async () => {
+    const historyOptions: HistoryOptions = {
+      ...(MINIMAL_CONFIG_MOCK as HistoryOptions),
+    };
```

Comment:
```suggestion
    const historyOptions = MINIMAL_CONFIG_MOCK as HistoryOptions;
```

There are several more occurrences in the file.

Reply:
@BioPhoton 

> There are several more occurrences in the file.

:top: I don't want to pollute PRs by commenting on every single occurrence.

Reply:
SRY! Resolved now. I also created a better mock object